projects/dsp-ui/src/lib/core/session.service.spec.ts

```diff
@@ -0,0 +1,155 @@
+import { async, TestBed } from '@angular/core/testing';
+import { KnoraApiConfig, MockUsers, UsersEndpointAdmin, AuthenticationEndpointV2, CredentialsResponse, ApiResponseData, ApiResponseError, LoginResponse } from '@dasch-swiss/dsp-js';
+import { of } from 'rxjs';
+import { DspApiConfigToken, DspApiConnectionToken } from './core.module';
+import { SessionService, Session } from './session.service';
+
+describe('SessionService', () => {
+    let service: SessionService;
+
+    const dspConfSpy = new KnoraApiConfig('http', 'localhost', 3333, undefined, undefined, true);
+
+    const dspConnSpy = {
+        admin: {
+            usersEndpoint: jasmine.createSpyObj('usersEndpoint', ['getUser'])
+        },
+        v2: {
+            auth: jasmine.createSpyObj('auth', ['checkCredentials', 'login'])
+        }
+    };
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [],
+            providers: [
+                {
+                    provide: DspApiConfigToken,
+                    useValue: dspConfSpy
+                },
+                {
+                    provide: DspApiConnectionToken,
+                    useValue: dspConnSpy
+                }
+            ]
+        });
+        service = TestBed.inject(SessionService);
+    }));
+
+    // mock localStorage
+    beforeEach(() => {
+        let store = {};
+
+        spyOn(localStorage, 'getItem').and.callFake(
+            (key: string): string => {
+                return store[key] || null;
+            }
+        );
+        spyOn(localStorage, 'removeItem').and.callFake(
+            (key: string): void => {
+                delete store[key];
+            }
+        );
+        spyOn(localStorage, 'setItem').and.callFake(
+            (key: string, value: string): string => {
+                return (store[key] = value as any);
+            }
+        );
+        spyOn(localStorage, 'clear').and.callFake(() => {
+            store = {};
+        });
+    });
+
+    beforeEach(() => {
+
+        const valuesSpy = TestBed.inject(DspApiConnectionToken);
+
+        (valuesSpy.admin.usersEndpoint as jasmine.SpyObj<UsersEndpointAdmin>).getUser.and.callFake(
+            () => {
+                console.log('getting a fake user');
+
+                const loggedInUser = MockUsers.mockUser();
+                return of(loggedInUser);
+            }
+        );
+
+        (valuesSpy.v2.auth as jasmine.SpyObj<AuthenticationEndpointV2>).checkCredentials.and.callFake(
```

Comment:
@tobiasschweizer this mocked method is never called when trying to run the test on line 135

Reply:
ok, I will have a look at this.

Reply:
I think I found out why the mock is not called. It is because inside the session service a new `KnoraApiConnection` is instantiated:

https://github.com/dasch-swiss/knora-ui-ng-lib/blob/5f175699b3534732946ab85f7695de4d7d29151b/projects/dsp-ui/src/lib/core/session.service.ts#L198-L206

I think instead the `KnoraApiConnection` provided in the constructor should be used. I don't understand why a new connection object has to be constructed.